enumerable.rb

```diff
@@ -0,0 +1,102 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    arr = to_a
+    while i < arr.length
+      yield(arr[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    (0...length).each do |i|
+      yield(self[i], i)
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_selection = []
+    my_each do |x|
+      new_selection.push(x) if yield(x)
+    end
+    new_selection
+  end
+
+  def my_all?
+    return to_enum(:my_all?) unless block_given?
+
+    new_selection = []
+    my_each do |n|
+      new_selection.push(n)
+      return false unless yield(n)
+    end
+    true
+  end
+
+  def my_any?
+    return to_enum(:my_any?) unless block_given?
+
+    arr = []
+    my_each do |x|
+      arr.push(x)
+      return true if yield(x)
+    end
+    false
+  end
+
+  def my_none?
+    return to_enum(:my_none?) unless block_given?
+
+    new_selection = []
+    my_each do |x|
+      new_selection.push(x)
+      return true if yield(x)
+    end
+    false
+  end
```

Comment:
- [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_none?

- [ ] when no block or argument is given returns true only if none of the collection members is true
```ruby
[nil, false, true, []].my_none? == [nil, false, true, []].none? => false
```
- Yours is retrieving the enumerable
![image](https://user-images.githubusercontent.com/52209067/78922444-c5586400-7a53-11ea-9ca7-621f6d20e0b7.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690365-cd4ed900-6f68-11ea-9933-f73b0091208d.png)

- [ ] when a class is passed as an argument returns true if none of the collection is a member of such class
```ruby
[1, 2.5, 'a', 9].my_none?(Integer) == [1, 2.5, 'a', 9].none?(Integer) => false
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/78922514-e456f600-7a53-11ea-91ef-0eea26fbcb64.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690539-130ba180-6f69-11ea-8bd0-6ca061ab05d2.png)

- [ ] when a Regex is passed as an argument returns true only if none of the collection matches the Regex
```ruby
%w[dog door rod blade].my_none?(/z/) == %w[dog door rod blade].none?(/z/) => true
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/78922570-018bc480-7a54-11ea-94e3-bbb2e883699b.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690722-65e55900-6f69-11ea-8881-9afe27d263dc.png)

- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
```ruby
[3,4,7,11].my_none?(3) == [3,4,7,11].none?(3) => false
```
- Yours is retrieving a ArgumentError
![image](https://user-images.githubusercontent.com/52209067/78922648-1cf6cf80-7a54-11ea-8210-8c75a00b4245.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690915-a9d85e00-6f69-11ea-81b3-7adb07b281ca.png)

[OPTIONAL]

I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:
```ruby
  def my_none?(arg = nil, &block)
    !my_any?(arg, &block)
  end
```
To make this work we have to use block.call(element) instead of yield(element) in my_any?